webapp/src/components/NameManager/NamedRange.tsx

```diff
@@ -0,0 +1,219 @@
+        console.log("DefinedName save failed", error);
+      }
+      onCreate?.();
+    }
+    setShowEditDelete(false);
+    toggleOptions();
+  };
+
+  const handleCancel = () => {
+    setReadOnly(true);
+    setShowEditDelete(false);
+    toggleOptions();
+    setNewName(name);
+    setNewScope(scope);
+    toggleShowNewName?.(); // if it's newName remove it from modal
+  };
+
+  const handleEdit = () => {
+    setReadOnly(false);
+    setShowEditDelete(true);
+    toggleOptions();
+  };
+
+  const handleDelete = () => {
+    try {
+      model.deleteDefinedName(newName, newScope);
+    } catch (error) {
+      console.log("DefinedName delete failed", error);
+    }
+    onDelete?.();
+  };
+
+  return (
+    <>
+      <StyledBox>
+        <StyledTextField
+          id="name"
+          variant="outlined"
+          size="small"
+          margin="none"
+          fullWidth
+          InputProps={{ readOnly: readOnly }}
+          error={nameError}
+          value={newName}
+          onChange={(event) => setNewName(event.target.value)}
+          onKeyDown={(event) => {
+            event.stopPropagation();
+          }}
+          onClick={(event) => event.stopPropagation()}
+        />
+        <StyledTextField
+          id="scope"
+          variant="outlined"
+          select
+          size="small"
+          margin="none"
+          fullWidth
+          InputProps={{ readOnly: readOnly }}
+          value={newScope ?? "global"}
+          onChange={(event) => {
+            event.target.value === "global"
+              ? setNewScope(undefined)
+              : setNewScope(+event.target.value);
+          }}
+        >
+          <MenuItem value={"global"}>
+            {t("name_manager_dialog.workbook")}
+          </MenuItem>
+          {worksheets.map((option, index) => (
+            <MenuItem key={option.name} value={index}>
+              {option.name}
+            </MenuItem>
+          ))}
+        </StyledTextField>
+        <StyledTextField
+          id="formula"
+          variant="outlined"
+          size="small"
+          margin="none"
+          fullWidth
+          InputProps={{ readOnly: readOnly }}
+          error={formulaError}
+          value={newFormula}
+          onChange={(event) => setNewFormula(event.target.value)}
+          onKeyDown={(event) => {
+            event.stopPropagation();
+          }}
+          onClick={(event) => event.stopPropagation()}
+        />
+
+        {showEditDelete ? (
+          // save cancel
+          <>
+            <IconButton
+              onClick={handleCreateUpdate}
+              sx={{ color: (theme) => theme.palette.success.main }}
+            >
+              <Check size={12} />
+            </IconButton>
+            <IconButton
+              onClick={handleCancel}
+              sx={{ color: (theme) => theme.palette.error.main }}
+            >
+              <X size={12} />
+            </IconButton>
+          </>
+        ) : (
```

Comment:
This is too much logic in a single function. You can split this two methods for active and non active ranges

Reply:
Splitted it!